### PR TITLE
feat: use new service account in escalate action

### PR DESCRIPTION
**Pull request by rgraber**
Issue: https://github.com/edx/edx-arch-experiments/issues/119

The user is recently created by IT and the scope of what we should use it for is not 100% determined but @nedbat and I determined that it makes sense to use it here.

### CODE REVIEW

File: .github/workflows/escalate.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
         uses: atlassian/gajira-login@v3
         env:
           JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
-          JIRA_USER_EMAIL: 'rgraber@edx.org'
+          JIRA_USER_EMAIL: 'productivitysysops@2u.com'
```
Comment (robrap) [inline .github/workflows/escalate.yml:16]: Can you provide a link to any existing (or new docs) that explains what productivitysysops@2u.com is and how it should be used? Please update the PR description. Thanks.

Comment (rgraber) [reply .github/workflows/escalate.yml:16]: There are currently no such docs as the user is new and IT has not put much information around it

Comment (robrap) [reply .github/workflows/escalate.yml:16]: If not a doc, I recommend we write at least a comment about where this email came from and how it is meant to be used. Could even just be a PR comment, but something to explain what you know now, but no one will know when they look at this in the future. The PR description is simply providing no context, including tickets/issues/etc.

**Review by robrap — APPROVED** [review]
Approving the change, but contingent on updating the PR with additional context for future us.

**Pull request merged**